src/undocumented_names.jl

```diff
@@ -0,0 +1,26 @@
+"""
+    test_undocumented_names(module::Module)
+
+Test that all public names in `module` have a docstring (including the module itself).
+
+!!! warning
+    For Julia versions before 1.11, this does not test anything.
+"""
```

Comment:
I would prefer seeing this warning in code with an `@warn` rather than a comment.

Reply:
That means adding a warning for every test below Julia 1.11, not sure people will like that. The fact that it does nothing on 1.10 can already be seen from the fact that the relevant `@testset` is empty

Reply:
I realize now that this implementation relies heavily on the "new" `public` keyword, which was not actually in the original issue (as it predated the keyword). Given that a main reason for introducing the `public` keyword was that previously something being documented was used to show it was part of the API, **I think the proposed implementation is a good idea**. But I'm not quite sure how to deal with the expectations towards <1.11 users. It _could_ help to change the name of the check to something like `test_public_documentation`, `test_documented_api` or something else that include the word "public" without being as verbose as `test_undocumented_public_names`.

Reply:
Given that `Docs.undocumented_names` is not defined on 1.10, I'm not sure we can do anything meaningful before 1.11 (unless that function is added to Compat.jl)